Pod/Classes/SEGAmplitudeIntegration.m

```diff
@@ -77,56 +77,61 @@ - (void)realTrack:(NSString *)event properties:(NSDictionary *)properties integr
-            }
-            id receipt = [properties objectForKey:@"receipt"];
-            if (!receipt || ![receipt isKindOfClass:[NSString class]]) {
-                receipt = nil;
-            }
-            NSLog(@"Number : %@", revenue);
-            [self.amplitude logRevenue:productId
-                              quantity:[quantity integerValue]
-                                 price:revenue
-                               receipt:receipt];
+        [self trackRevenue:properties andRevenue:revenue];
+    }
+}
+
+- (void)trackRevenue:(NSDictionary *)properties andRevenue:(NSNumber *)revenue
+{
+    // Use logRevenueV2 with revenue properties.
+    if ([(NSNumber *)[self.settings objectForKey:@"useLogRevenueV2"] boolValue]) {
+        id price = [properties objectForKey:@"price"];
+        id quantity = [properties objectForKey:@"quantity"];
+
+        // if no price fallback to using revenue
+        if (!price || ![price isKindOfClass:[NSNumber class]]) {
+            price = revenue;
+            quantity = [NSNumber numberWithInt:1];
+        } else if (!quantity || ![quantity isKindOfClass:[NSNumber class]]) {
+            quantity = [NSNumber numberWithInt:1];
+        }
+
+        [[self.amprevenue setPrice:price] setQuantity:[quantity integerValue]];
+        id productId = [properties objectForKey:@"productId"] ?: [properties objectForKey:@"product_id"];
+        if (productId && [productId isKindOfClass:[NSString class]] && ![productId isEqualToString:@""]) {
+            [self.amprevenue setProductIdentifier:productId];
+        }
+
+        //Receipt is meant to be of type NSData
+        id receipt = [properties objectForKey:@"receipt"];
+        if (receipt && [receipt isKindOfClass:[NSString class]] && ![receipt isEqualToString:@""]) {
+            [self.amprevenue setReceipt:receipt];
+        }
+        id revenueType = [properties objectForKey:@"revenueType"] ?: [properties objectForKey:@"revenue_type"];
+        if (revenueType && [revenueType isKindOfClass:[NSString class]] && ![revenueType isEqualToString:@""]) {
+            [self.amprevenue setRevenueType:revenueType];
+        }
+        NSLog(@"Price : %@, Quantity : %@", price, quantity);
+        [self.amplitude logRevenueV2:self.amprevenue];
+    } else {
+        // fallback to logRevenue v1
+        id productId = [properties objectForKey:@"productId"] ?: [properties objectForKey:@"product_id"];
+        if (!productId || ![productId isKindOfClass:[NSString class]]) {
+            productId = nil;
+        }
+        id quantity = [properties objectForKey:@"quantity"];
+        if (!quantity || ![quantity isKindOfClass:[NSNumber class]]) {
+            quantity = [NSNumber numberWithInt:1];
+        }
+        id receipt = [properties objectForKey:@"receipt"];
+        if (!receipt || ![receipt isKindOfClass:[NSString class]]) {
+            receipt = nil;
         }
+        NSLog(@"Number : %@", revenue);
```

Comment:
This is form before, but what does this doo?

Reply:
I removed it in https://github.com/segment-integrations/analytics-ios-integration-amplitude/pull/35